### PR TITLE
Fix potential security issues in origin validation

### PR DESCRIPTION
**Pull request by FrederikBolding**
[ch8972]

### CODE REVIEW

File: packages/signer/src/api/ws.sagas.ts [hunk 1]
```diff
@@ -146,15 +154,41 @@ export function* handleRequest({ socket, request, data }: WebSocketMessage) {
 
   const { signature, publicKey, ...jsonRpcRequest } = fullRequest;
 
-  const permissions: Permission[] = yield select(getPermissions);
-  const origin = request.headers.origin && new URL(request.headers.origin).host;
+  const origin = getOriginHost(request);
+  // Dont serve requests without origin
+  if (origin === undefined || origin === null) {
```
Comment (Mrtenz) [inline packages/signer/src/api/ws.sagas.ts:159]:
```suggestion
  if (origin == null) {
```
or just
```suggestion
  if (!origin) {
```

File: packages/signer/src/utils/hashing.ts [hunk 2]
```diff
@@ -17,6 +17,10 @@ export const signRequest = async (data: JsonRPCRequest, privateKey: string) => {
 };
 
 export const verifyRequest = async (sig: string, data: JsonRPCRequest, publicKey: string) => {
-  const hash = await hashRequest(data);
-  return verify(sig, hash, publicKey);
+  try {
+    const hash = await hashRequest(data);
+    return await verify(sig, hash, publicKey);
```
Comment (Mrtenz) [inline packages/signer/src/utils/hashing.ts:22]: `await` is redundant when using `return`.

**Review by Mrtenz — CHANGES_REQUESTED** [review]

File: packages/signer/src/utils/hashing.ts (hunk 2 above)
Comment (FrederikBolding) [reply packages/signer/src/utils/hashing.ts:22]: Not when trying to catch the promise rejection!

File: packages/signer/src/api/ws.sagas.ts (hunk 1 above)
Comment (FrederikBolding) [reply packages/signer/src/api/ws.sagas.ts:159]: I would prefer being explicit in this case if it is fine with you.

Comment (Mrtenz) [reply packages/signer/src/api/ws.sagas.ts:159]: Then using the first option should be fine, right?

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**